# ABOUT TEAM:

- 18120517 Nguyễn Công Bình Phương
- 18120514 Phan Thiên Phúc

# LỜI MỞ ĐẦU

***Bạn có biết rằng***:
- Hơn 700 000 người chết do tự tử mỗi năm. 
- Cứ mỗi vụ tự tử lại có thêm nhiều người cố gắng tự tử. Nỗ lực tự tử trước đó là yếu tố nguy cơ quan trọng nhất dẫn đến tự tử trong dân số nói chung. 
- Tự tử là nguyên nhân thứ tư gây tử vong ở lứa tuổi 15-19. 
- 77% số vụ tự tử trên toàn cầu xảy ra ở các nước có thu nhập thấp và trung bình. Uống thuốc trừ sâu, treo cổ và cầm súng là một trong những phương pháp tự sát phổ biến nhất trên toàn cầu.

<a href='https://www.who.int/news-room/fact-sheets/detail/suicide'> nguồn thông tin </a>

Phía trên là những thông tin xác thực từ tổ chức nổi tiếng của thế giới WHO (World Healthy Organization). Rõ ràng vấn đề tự tử là một vấn đề nghiêm trọng của thế giới. Tự tử là một vấn đề bi kịch đến từ yếu tố tâm lý của người có ý định tự sát, tự kết thúc mạng sống của bản thân vì một số lý do nào đó: áp lực tinh thần, lạc lõng, mất phương hướng trong cuộc sống, bị đe dọa, ...

Tự sát không chỉ tước đi mạng sống của nạn nhân mà còn gây tổn hại cho người thân, gia đình và xã hội. Khi ta mất đi một người thân, một người bạn chỉ vì họ tự tử thì đó quả thật là một điều đau lòng, sự dằn vặt trong ta. Một người tự tử có thể trở thành một hiệu ứng lan truyền như một cách thức giải quyết cho những người cảm thấy bế tắc cho cuộc sống.

***Vậy nên việc phân tích dữ liệu về vấn đề tự tử này sẽ mang đến ý nghĩa gì?***

- Đầu tiên là nói về dữ liệu này: dữ liệu này là dữ liệu về vấn đề tự tử của 101 quốc gia từ năm 1985 đến năm 2016. Dữ liệu này bao gồm một số thông tin hữu ích liên quan như: giới tính, tuổi, số lượng tự tử, mật độ dân số, gdp của năm, gdp bình quân đâu người, thế hệ.

- Từ dữ liệu này ta sẽ phân tích được một số insight về sự liên quan giữa các thuộc tính trong dữ liệu với việc tự tử: ví dụ như ở độ tuổi nào có số lượng tự tử nhiều. Với mỗi giới tính, từng năm, kinh tế thì ảnh hưởng như thế nào đến tự tử

- Vậy từ các insight có thể có ý nghĩa như thế nào?
    Đôi khi trong cuộc sống, ở mỗi giai đoạn ta sẽ trải qua những áp lực khác nhau. Khi còn độ tuổi học sinh, ta áp lực về chuyện học hành, bố mẹ kì vọng. Ở độ tuổi đi làm ta áp lực về tiền bạc, áp lực công việc. Khi có gia đình ta áp lực về việc nuôi dạy con cái, xây dựng hạnh phúc gia đình, chuyện tiền nong, học hành của con cái, ... Vậy nên với mỗi giai đoạn trưởng thành, nhiều lúc bế tắc trong cuộc sống, có lẽ thứ ta cần là một sự đồng cảm. Vậy nên việc phân tích dữ liệu này là một điều cần thiết, đôi khi nhìn vào số liệu này ta nhận ra được một việc rằng đó là có rất nhiều người đều như chúng ta, họ cũng trải qua bế tắc trong cuộc sống.
    
- Đặc biệt trong tình hình dịch bênh covid phức tạp như hiện nay, nhiều người thất nghiệp, cuộc sống hạn chế quanh quẩn trong nhà, học sinh không được đến trường, các ca tử vong do dịch. Cuộc sống như hiện tại gây nên rất nhiều âu lo cho mọi người, thất nghiệp, cô đơn, quây quẩn xung quên 4 bức tường, cày netflix chờ đến khi cuộc sống ổn định rồi tính tiếp. Có lẽ không ít người rơi vào trầm cảm và nhiều khi nghĩ đến cái chết...

.
.
.
    
    "A useless life is an early death" - Johann Wolfgang von Goethe
                           
    
    "Người chết đã đi rồi nhưng nỗi đau của những người ở lại có lẽ sẽ kéo dài vô tận." - Tình Yêu Pha Lê -- Tuyết Ảnh Sương Hồn

# PHÂN TÍCH DỮ LIỆU

## ABOUT DATA
- **Source**: https://www.kaggle.com/russellyates88/suicide-rates-overview-1985-to-2016

- **Content**

    This compiled dataset pulled from four other datasets linked by time and place, and was built to find signals correlated to increased suicide rates among different cohorts globally, across the socio-economic spectrum.

- **References**
    
    United Nations Development Program. (2018). Human development index (HDI). Retrieved from http://hdr.undp.org/en/indicators/137506

    World Bank. (2018). World development indicators: GDP (current US$) by country:1985 to 2016. Retrieved from http://databank.worldbank.org/data/source/world-development-indicators#

    [Szamil]. (2017). Suicide in the Twenty-First Century [dataset]. Retrieved from https://www.kaggle.com/szamil/suicide-in-the-twenty-first-century/notebook

    World Health Organization. (2018). Suicide prevention. Retrieved from http://www.who.int/mental_health/suicide-prevention/en/

- **Inspiration**
    
    Suicide Prevention.
    
- **Decription of feature**
    + Country: Column representing countries (string)
    + Year: Column representing years (integer)
    + Sex: Column representing genres (string)
    + Age: Column representing ages (string)
    + Suicides_no: Column representing the number of suicides (integer)
    + Population: Column representing the population (integer)
    + Suicides / 100k pop: Column that represents the number of suicides per 100 thousand inhabitants (float)
    + Country-year: Column that represents (string)
    + HDI for year: Column that represents the HDI in each year (float)
    + Gdp_for_year($): Column representing GDP in each year (string)
    + Gdp_per_capita($): Column representing GDP per capita (float)
    + Generation: Column that represents the generation (string)



Đầu tiên ta sẽ xem tổng quan về dữ liệu, có bao nhiều dòng, dữ liệu có thiếu không, thông tin về các cột, phân bố của các thuộc tính, ...

## TỔNG QUAN

In [2]:
df = pd.read_csv('master.csv')
df.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

   suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0               6.71  Albania1987           NaN      2,156,624,900   
1               5.19  Albania1987           NaN      2,156,624,900   
2               4.83  Albania1987           NaN      2,156,624,900   
3               4.59  Albania1987           NaN      2,156,624,900   
4               3.28  Albania1987           NaN      2,156,624,900   

   gdp_per_capita ($)       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27820 entries, 0 to 27819
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   country             27820 non-null  object 
 1   year                27820 non-null  int64  
 2   sex                 27820 non-null  object 
 3   age                 27820 non-null  object 
 4   suicides_no         27820 non-null  int64  
 5   population          27820 non-null  int64  
 6   suicides/100k pop   27820 non-null  float64
 7   country-year        27820 non-null  object 
 8   HDI for year        8364 non-null   float64
 9    gdp_for_year ($)   27820 non-null  object 
 10  gdp_per_capita ($)  27820 non-null  int64  
 11  generation          27820 non-null  object 
dtypes: float64(2), int64(4), object(6)
memory usage: 2.5+ MB


In [9]:
df.isnull().sum() / df.shape[0]

country               0.000000
year                  0.000000
sex                   0.000000
age                   0.000000
suicides_no           0.000000
population            0.000000
suicides/100k pop     0.000000
country-year          0.000000
HDI for year          0.699353
 gdp_for_year ($)     0.000000
gdp_per_capita ($)    0.000000
generation            0.000000
dtype: float64

- Dữ liệu có tổng cộng 12 cột, 27820 dòng. Trong đó cột duy nhất một cột thiếu dữ liệu là `HDI for year`, tỉ lệ thiếu đến 69.99%. Cột này mang ý nghĩa là chỉ số HDI trong năm đó, chỉ số HDI là chỉ số phát triển con người

- Ta thấy có 5 cột thuộc kiểu object: country, sex, age, coutry-year, generation. Ta xem thử phân bố các nhãn thuộc các thuộc tính này như thế nào

In [15]:
obj_cols = ['country', 'sex', 'age', 'country-year', 'generation']

for obj_col in obj_cols:
    print(f'\n\n-----{obj_col}-----')
    print(df[obj_col].value_counts())
    



-----country-----
Iceland                   382
Netherlands               382
Mauritius                 382
Austria                   382
Argentina                 372
                         ... 
Bosnia and Herzegovina     24
Cabo Verde                 12
Dominica                   12
Macau                      12
Mongolia                   10
Name: country, Length: 101, dtype: int64


-----sex-----
female    13910
male      13910
Name: sex, dtype: int64


-----age-----
15-24 years    4642
55-74 years    4642
75+ years      4642
25-34 years    4642
35-54 years    4642
5-14 years     4610
Name: age, dtype: int64


-----country-year-----
Luxembourg2009     12
Denmark2011        12
Ukraine2001        12
Argentina2007      12
Argentina1994      12
                   ..
Armenia2016        10
Netherlands2016    10
Austria2016        10
Iceland2016        10
Croatia2016        10
Name: country-year, Length: 2321, dtype: int64


-----generation-----
Generation X       6408
Silent          

- Ta tổng hợp được như sau:
    + `country`: Có tổng cộng 101 quốc gia, nhưng số lượng các quốc gia xuất hiện không đều nhau (có lẽ liên quan đến cột year)
    + `sex`: có hai giới tính (male, female). Hmm, khá cơ bản, nếu có thêm giới tính thứ 3 thì có lẽ việc phân tích sẽ hay hơn
    + `age`: gồm 6 nhãn, chia như sau: 5-14, 15-24, 25-34, 35-54, 55-74, 75+. Các nhãn này phân bố đều nhau. Chỉ só duy nhất nhãn 5-14 thì ít hơn các nhãn còn lại
    + `country-year`: cột này chỉ đơn giản là sự kết hợp giữa year và country
    + `generation`: gồm 6 nhãn, phân bố không đều nhau, về ý nghĩa của các nhãn thì trong mô tả dữ liệu không đề cập nhưng ta có thể search google. Sau khi search thì ta có được thông tin của các nhãn như sau:
        
        
        
        
        + G.I. Generation: Thế hệ vĩ đại nhất, còn được gọi là G.I. Thế hệ và Thế hệ Chiến tranh thế giới thứ hai, là nhóm nhân khẩu học theo sau Thế hệ Mất tích và trước Thế hệ Im lặng. Thế hệ này thường được định nghĩa là những người sinh từ năm 1901 đến năm 1927
        
        + Silent: (được sinh ra trong khoảng 1925 đến 1945) phản ứng lại cuộc Đại khủng hoảng kinh tế và Thế chiến thứ hai bằng cách chăm chỉ làm việc và giữ im lặng trước những cuộc phản kháng hay quan điểm chính trị của các thế hệ trước.
        
        + Boomers: (thế hệ được sinh ra trong thời kỳ bùng nổ dân số: 1944 đến 1965). Lớn lên trong một thế giới đầy những thành phố đang phát triển và gia đình đông con, họ tin rằng chỉ cần chăm chỉ làm việc thì sẽ có cuộc sống tốt hơn. Nhưng họ vất vả nhiều mà hưởng chẳng bao nhiêu, chủ yếu là lo cái ăn cái mặc và mong con cái học hành đỗ đạt để có thể đổi đời.
        
        + Generation X: Thế hệ X (tiếng Anh: Generation X, viết tắt: Gen X) chỉ những người sinh ra giữa những năm 1960 đến đầu những năm 1980. Khoảng 68% CEO các tập đoàn trong danh sách Fortune 500 thuộc thế hệ X.
        
        + Millenials: (gen Y)Những bạn trẻ ra đời cuối những năm 80 và đầu những năm 90, thế hệ chịu tác động trực tiếp của sự phát triển công nghệ. Mọi thứ đều nhanh gọn lẹ, có tiện có lợi nhưng cũng có hậu quả: thế hệ này muốn được mọi thứ ngay lập tức, thiếu kiên nhẫn và dễ dao động.
        
        + Generation Z: (1995 đến 2009) Những ai sinh từ những năm cuối thế kỷ 20 đến đầu thế kỷ 21: thiết bị thông minh, mạng xã hội, công nghệ cao, sống nhanh sống ảo, chịu nhiều áp lực cạnh tranh về vật chất và danh vọng. Nhiều em được gia đình bảo bọc quá mức nên thiếu kỹ năng sống và lệch lạc trong hành xử. Thế hệ này học nhiều, học miết nhưng phần lớn là học giùm kỳ vọng của mẹ, của cha.

In [14]:
df.head(12)

country  year     sex          age  suicides_no  population  \
0   Albania  1987    male  15-24 years           21      312900   
1   Albania  1987    male  35-54 years           16      308000   
2   Albania  1987  female  15-24 years           14      289700   
3   Albania  1987    male    75+ years            1       21800   
4   Albania  1987    male  25-34 years            9      274300   
5   Albania  1987  female    75+ years            1       35600   
6   Albania  1987  female  35-54 years            6      278800   
7   Albania  1987  female  25-34 years            4      257200   
8   Albania  1987    male  55-74 years            1      137500   
9   Albania  1987  female   5-14 years            0      311000   
10  Albania  1987  female  55-74 years            0      144600   
11  Albania  1987    male   5-14 years            0      338200   

    suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0                6.71  Albania1987           NaN      2,156,624,900   
1                5.19  Albania1987           NaN      2,156,624,900   
2                4.83  Albania1987           NaN      2,156,624,900   
3                4.59  Albania1987           NaN      2,156,624,900   
4                3.28  Albania1987           NaN      2,156,624,900   
5                2.81  Albania1987           NaN      2,156,624,900   
6                2.15  Albania1987           NaN      2,156,624,900   
7                1.56  Albania1987           NaN      2,156,624,900   
8                0.73  Albania1987           NaN      2,156,624,900   
9                0.00  Albania1987           NaN      2,156,624,900   
10               0.00  Albania1987           NaN      2,156,624,900   
11               0.00  Albania1987           NaN      2,156,624,900   

    gdp_per_capita ($)       generation  
0                  796     Generation X  
1                  796           Silent  
2                  796     Generation X  
3                  796  G.I. Generation  
4                  796          Boomers  
5                  796  G.I. Generation  
6                  796           Silent  
7                  796          Boomers  
8                  796  G.I. Generation  
9                  796     Generation X  
10                 796  G.I. Generation  
11                 796     Generation X

Ở trên thì ta xem được 12 dòng đầu của dữ liệu, thì đây cũng là cách mà dữ liệu phân bố. Để hiểu hơn thì mình xin giải thích như sau:
- Với mỗi quốc gia tường ứng với mỗi năm thì sẽ bao gồm 2 giới tính: male, female. Tương ứng với mỗi giơi tính thì sẽ là các age (5 nhãn hoặc 6 nhãn). Nên tương ứng với mỗi quốc gia và năm thì bao gồm $2*5=10$ hoặc $2*6=12$ dòng dữ liệu.

- Vậy nên việc phân tích ý nghĩa tương quan khác nhau giữa các thuộc tính sẽ dựa vào sự chênh lệch, tương quan giữa các thuộc tính số (số lượng tự tử, chỉ số gdp, mật độ dân số, ...)